### PR TITLE
EASY-2285 sort files lexicographically

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes EASY-2285 sort files lexicographically

#### When applied it will
* 
* 
* 

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.deposit/DataFiles.scala [hunk 1]
```diff
@@ -69,6 +69,7 @@ case class DataFiles(bag: DansBag) extends DebugEnhancedLogging {
         .withFilter(_._1.isChildOf(parentPath))
         .map((toFileInfo _).tupled)
         .toSeq
+        .sortBy(fileInfo => (fileInfo.dirpath, fileInfo.dirpath))
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/DataFiles.scala]: you're sorting by `dirpath` twice???

Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.deposit/DataFiles.scala]: Somehow managed to push only half of the changes.

File: src/main/scala/nl.knaw.dans.easy.deposit/DataFiles.scala [hunk 2]
```diff
@@ -24,6 +24,7 @@ import nl.knaw.dans.bag.DansBag
 import nl.knaw.dans.easy.deposit.Errors.NoSuchFileInDepositException
 import nl.knaw.dans.easy.deposit.docs.FileInfo
 import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.joda.time.DateTime
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/DataFiles.scala]: Unused import?
```suggestion
```

File: src/main/scala/nl.knaw.dans.easy.deposit/DataFiles.scala [hunk 3]
```diff
@@ -65,10 +66,12 @@ case class DataFiles(bag: DansBag) extends DebugEnhancedLogging {
     }
 
     def convert(items: Map[File, String]) = {
+      implicit val pathOrdering: Ordering[Path] = Ordering.fromLessThan[Path](_.compareTo(_) < 0)
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/DataFiles.scala]: From what I read here, `Path` is comparable. Hence you can derive the `Ordering` from that. You don't need this workaround with `fromLessThan`.

File: src/test/scala/nl.knaw.dans.easy.deposit/DataFilesSpec.scala [hunk 4]
```diff
@@ -99,24 +99,29 @@ class DataFilesSpec extends TestSupportFixture {
     }
   }
 
-  "fileInfoSeq" should "return the proper number of files" in {
+  "list" should "return files in lexicographical order" in {
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/DataFilesSpec.scala]: not sure if it's a lexicographical order anymore??? Up to you to figure that out 😉

Comment (jo-pol) [reply src/test/scala/nl.knaw.dans.easy.deposit/DataFilesSpec.scala]: Curious which order will emerge from travis once the dependency problem is fixed.

@lindareijnhoudt 

At least the files are grouped per folder. Is that enough?

File: src/test/scala/nl.knaw.dans.easy.deposit/DataFilesSpec.scala [hunk 5]
```diff
@@ -99,24 +100,51 @@ class DataFilesSpec extends TestSupportFixture {
...
+       * Compares two abstract paths lexicographically. The ordering defined by
+       * this method is provider specific, and in the case of the default
+       * provider, platform specific. This method does not access the file system
+       * and neither file is required to exist.
+       */
+      /* Observation:
+       *
+       * depth first
+       * at one level files an folders are mixed
+       * a sequence of digits seems to be treated as a single character (unlike an ls command) but not numerically
```
Comment (jo-pol) [inline src/test/scala/nl.knaw.dans.easy.deposit/DataFilesSpec.scala]: An experiment with iTerm on mac:
```
tree
.
|-- 1.txt
|-- f
|   `-- some.txt
|-- f#b
|-- f.xy
|   `-- some.txt
`-- f1
```

File: src/test/scala/nl.knaw.dans.easy.deposit/DataFilesSpec.scala (hunk 4 above)
Comment (rvanheest) [reply src/test/scala/nl.knaw.dans.easy.deposit/DataFilesSpec.scala]: Given the paths below, I would expect the following order (based on the tree command):

```
./#/1.txt
./.hidden
./1.txt
./folder1/3.txt
./folder1/5.txt
./folder1/b/x.txt
./folder1#b/x.txt
./folder1.x/y.txt
./folder11/4.txt
./folder1y/x.txt
./folder2/4.txt
./foo.txt
```

**Pull request merged**